tigramite/toymodels/context_model.py

```diff
@@ -0,0 +1,221 @@
+            noises_tc = None
+        shifted_links_tc = shift_link_entries(links_tc, -self.N)
+        data_tc, nonstat_tc = toys.structural_causal_process(shifted_links_tc, T=T, noises=noises_tc,
+                                                             seed=MT19937(seed))
+        data_tc = {i: data_tc[:, i - self.N] for i in links_tc.keys()}
+
+        data_tc_list = self.constant_over_space(data_tc, M)
+        return data_tc_list, np.any(nonstat_tc)
+
+    def _generate_spatial_context_data(self, links_sc, T, M, shift, seed):
+        """
+        Helper Function to generate data for the spatial context nodes. It essentially is a
+        wrapper around toys.structural_causal_process to generate data that is random across datasets
+        but constant across time.
+        """
+        shifted_links_sc = shift_link_entries(links_sc, -shift)
+        if self.noises is not None:
+            noises_sc = [self.noises[key] for key in links_sc.keys()]
+        else:
+            noises_sc = None
+        data_sc, nonstat_sc = toys.structural_causal_process(shifted_links_sc, T=M, noises=noises_sc,
+                                                             seed=MT19937(seed))
+
+        data_sc_list = self.constant_over_time(data_sc, T, M, shift)
+        return data_sc_list, np.any(nonstat_sc)
+
+    def generate_data(self, M, T):
+        """
+        Generates M datasets of time series generated from a joint structural causal model over different spatial
+        and temporal contexts.
+
+        Returns
+        ----------
+        data : dictionary with array-like values
+            Datasets generated from this process, each dataset has the shape (T, N).
+        nonvalid : bool
+            Indicates whether data has NaNs or infinities.
+        """
+
+        links = {**self.links_tc, **self.links_sc, **self.links_sys}
+
+        K_time = len(self.links_tc.keys())
+
+        data = {}
+        nonstationary = []
+
+        child_seeds = self.seed.spawn(3)
+
+        # first generate data for temporal context nodes
+        data_tc_list, nonstat_tc = self._generate_temporal_context_data(self.links_tc, T, M, child_seeds[0])
+
+        # generate spatial context data (constant in time)
+        data_sc_list, nonstat_sc = self._generate_spatial_context_data(self.links_sc,
+                                                                       T, M,
+                                                                       K_time + self.N,
+                                                                       child_seeds[1])
+        for m in range(M):
+            data_context = dict(data_tc_list[m])
+            data_context.update(data_sc_list[m])
+
+            # generate system data that varies over space and time
+            data_m, nonstat = toys.structural_causal_process(links, T=T, intervention=data_context,
+                                                             seed=MT19937(child_seeds[2]), noises=self.noises)
```

Comment:
This part with MT19937: Can you use standard random_state stuff as done in other parts of tigramite?